bundles/org.openhab.binding.sagercaster/src/main/java/org/openhab/binding/sagercaster/internal/handler/SagerCasterHandler.java

```diff
@@ -0,0 +1,269 @@
+                values.entrySet().removeIf(map -> map.getKey() <= eldestKey.get());
+            }
+        }
+
+        public @Nullable T getAgedValue() {
+            return agedValue;
+        }
+    }
+
+    private final ExpiringMap<QuantityType<Pressure>> pressureCache = new ExpiringMap<>();
+    private final ExpiringMap<QuantityType<Temperature>> temperatureCache = new ExpiringMap<>();
+    private final ExpiringMap<QuantityType<Angle>> bearingCache = new ExpiringMap<>();
+
+    public SagerCasterHandler(Thing thing, WindDirectionStateDescriptionProvider stateDescriptionProvider) {
+        super(thing);
+        this.stateDescriptionProvider = stateDescriptionProvider;
+    }
+
+    @Override
+    public void initialize() {
+        String location = (String) getConfig().get(CONFIG_LOCATION);
+        int observationPeriod = ((BigDecimal) getConfig().get(CONFIG_PERIOD)).intValue();
+        String latitude = location.split(",")[0];
+        sagerWeatherCaster.setLatitude(Double.parseDouble(latitude));
+        long period = TimeUnit.SECONDS.toMillis(observationPeriod);
+        pressureCache.setObservationPeriod(period);
+        bearingCache.setObservationPeriod(period);
+        temperatureCache.setObservationPeriod(period);
+        defineWindDirectionStateDescriptions();
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    private void defineWindDirectionStateDescriptions() {
+        List<StateOption> options = new ArrayList<>();
+        String[] directions = sagerWeatherCaster.getUsedDirections();
+        for (int i = 0; i < directions.length; i++) {
+            int secondDirection = i < directions.length - 1 ? i + 1 : 0;
+            String windDescription = directions[i] + " or " + directions[secondDirection] + " winds";
+            options.add(new StateOption(String.valueOf(i + 1), windDescription));
+        }
+
+        options.add(new StateOption("9", "Shifting / Variable winds"));
+        stateDescriptionProvider.setStateOptions(new ChannelUID(getThing().getUID(), CHANNEL_WINDFROM), options);
+        stateDescriptionProvider.setStateOptions(new ChannelUID(getThing().getUID(), CHANNEL_WINDTO), options);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            postNewForecast();
+        } else {
+            String id = channelUID.getId();
+            switch (id) {
+                case CHANNEL_CLOUDINESS:
+                    logger.debug("Octa cloud level changed, updating forecast");
+                    if (command instanceof QuantityType) {
+                        @SuppressWarnings("unchecked")
+                        QuantityType<Dimensionless> cloudiness = (QuantityType<Dimensionless>) command;
+                        new Thread(() -> {
+                            sagerWeatherCaster.setCloudLevel(cloudiness.intValue());
+                            postNewForecast();
+                        }).start();
```

Comment:
Please don't create a new thread for this. Instead use the scheduler available to every handler.
```suggestion
                        scheduler.submit(() -> {
                            sagerWeatherCaster.setCloudLevel(cloudiness.intValue());
                            postNewForecast();
                        });
```